Robot2018/src/org/usfirst/frc/team199/Robot2018/Robot.java

```diff
@@ -48,12 +48,18 @@
 	SendableChooser<Position> posChooser = new SendableChooser<Position>();
 	Map<String, SendableChooser<Strategy>> stratChoosers = new HashMap<String, SendableChooser<Strategy>>();
 	String[] fmsPossibilities = {"LL", "LR", "RL", "RR"};
+		
+	public static SmartDashboardInterface sd = new SmartDashboardInterface() {
+		public double getConst(String key, double def) {
+			if (!SmartDashboard.containsKey("Const/" + key)) {
+				SmartDashboard.putNumber("Const/" + key, def);
+			}
+			return SmartDashboard.getNumber("Const/" + key, def);
+		}		
+	};
 
 	public static double getConst(String key, double def) {
-		if (!SmartDashboard.containsKey("Const/" + key)) {
-			SmartDashboard.putNumber("Const/" + key, def);
-		}
-		return SmartDashboard.getNumber("Const/" + key, def);
+		return sd.getConst(key, def);
```

Comment:
is the getConst() method even necessary now? and was moving the stuff into the sd interface approved? just making sure, nothing personal

Reply:
this is not mine, it might be part of Corvin's test though.

Reply:
I left the Robot.getConst() method to maintain compatibility to other parts of the code and avoid excessive refactoring without existing unit tests in place.  The functionality is unchanged.  However, the main smart dashboard constant retrieval flow is through the interface and Robot.sd implementation.  Sorry for not getting this pre-approved, but I had discussed with @lhmcgann and @brettle the need to separate the dashboard logic from the Robot class to facilitate unit testing.